packages/@ionic/cli/src/lib/hooks.ts

```diff
@@ -97,6 +104,25 @@ export abstract class Hook {
 
     debug(`Could not load hook function ${strong(p)}: %o not a function`, module);
   }
+
+  private generateCTXEnvironment(input: HookInput, path: string[] = []): NodeJS.ProcessEnv {
+    let environment: NodeJS.ProcessEnv = {};
+
+    for (const [key, value] of Object.entries(input)) {
+      if (typeof value === 'object') {
+        environment = {
+          ...environment,
+          ...this.generateCTXEnvironment(value, [...path, key]),
+        };
+      } else {
+        const name = [path, key].join('_');
```

Comment:
Is this supposed to be this?

```suggestion
        const name = [...path, key].join('_');
```